pipeline/stop_places/sql/stop_places.sql

```diff
@@ -605,16 +605,70 @@ $$
 LANGUAGE plpgsql IMMUTABLE STRICT;
 
 
+/*
+ * Create an aggregate function that combines multiple jsonb objects into one.
+ * This is used to combine the tags of multiple elements into one jsonb object.
+ * The pgsql function 'jsonb_object_agg' does not allow the input of jsonb objects.
+ * 'jsonb_agg' combines the objects into an array, which is not what we want.
+ * See: https://stackoverflow.com/questions/57249804/combine-multiple-json-rows-into-one-json-object-in-postgresql
+ */
+CREATE OR REPLACE AGGREGATE jsonb_combine(jsonb) 
+(
+    SFUNC = jsonb_concat(jsonb, jsonb),
+    STYPE = jsonb
+);
+
+
 /*********
  * QUAYS *
  *********/
 
+ /*
+  * Create a view where platforms with the same IFOPT are merged into one row.
+  * See: https://github.com/OPENER-next/osm2vdv462/issues/8
+  * Only osm elements that intersect with each other are merged. This is true for:
+  *  - touching or overlapping platforms mapped as ways or areas
+  *  - nodes that lie within the polygon or on the border of a area
+  *  - nodes that lie within a way (does not have to be part of the way)
+  * Only the osm_id with the highest value is kept for reference.
+  * Currently the osm_type is always 'W' (way).
+  *   This is because it's not likely that two nodes with the same IFOPT are overlapping and therefore will be merged.
+  * The tags of the elements will be merged.
+  *   If there is a key that has different values in the platforms, the value of the last platform is kept.
+  */
+CREATE OR REPLACE VIEW platforms_merged AS (
+  SELECT
+    'W'::CHAR as osm_type,
+    MAX(p3.osm_id) as osm_id,
+    p3."IFOPT",
+    jsonb_combine(tags) as tags,
+    ST_Union(p3.geom) as geom
+  FROM (
+    SELECT p1.* FROM platforms p1
+    JOIN platforms p2 ON p1."IFOPT" = p2."IFOPT"
+    WHERE ST_Intersects(p1.geom, p2.geom) AND NOT ST_Equals (p1.geom, p2.geom)
+  ) as p3
+	GROUP BY p3."IFOPT"
+);
+
+
+/*
+ * Create view that contains all platforms and replaces the splitted platforms with the merged ones.
+ * This is done by using a UNION of the merged platforms and the platforms that are not merged.
+ */
+CREATE OR REPLACE VIEW platforms_union AS (
+SELECT * from platforms_merged
+UNION
+SELECT * from platforms WHERE platforms."IFOPT" not in (SELECT "IFOPT" from platforms_merged)
+);
+
+
```

Comment:
Always assuming type "W" and combining it with `MAX(p3.osm_id)` looks error prone to me.
Since there are more nodes then ways the ids are usually higher so in most cases if a node and a way gets merged you will actually end up with the node id here. But even if this wasn't the case you can still end up with wrong id to type matching since platforms can also be relations.

At first I thought the following solution would be the best one, however I think it won't work if a platform is split into more then 2 elements.

```SQL
 -- ST_Normalize is important to make DISTINCT work, because the merging geometries a to b and from b to a produces different geometries (different orders or directions)

SELECT DISTINCT ON ("IFOPT", geom)
    p1.osm_type,
    p1.osm_id,
    p1."IFOPT" AS "IFOPT",
    COALESCE(jsonb_concat(p2.tags, p1.tags), p1.tags) as tags
    COALESCE(ST_Normalize(ST_Union(p1.geom, p2.geom)), p1.geom) AS geom,
FROM platforms p1 
LEFT JOIN platforms p2
    ON (p1.osm_type != p2.osm_type OR p1.osm_id != p2.osm_id)
    AND p1."IFOPT" = p2."IFOPT"
    AND ST_Intersects(p1.geom, p2.geom)
```

So I thought we require the GROUP BY class to solve this problem. However I think this can't be the solution either and suffers from the same problem.

```SQL
SELECT
    p1."IFOPT" AS "IFOPT",
    ST_Union(p1.geom) AS geom,
    jsonb_combine(p1.tags) AS tags
FROM platforms p1 
LEFT JOIN platforms p2
    ON (p1.osm_type != p2.osm_type OR p1.osm_id != p2.osm_id)
    AND p1."IFOPT" = p2."IFOPT"
    AND ST_Intersects(p1.geom, p2.geom)
GROUP BY
  p1."IFOPT",
  COALESCE(ST_Normalize(ST_Union(p1.geom, p2.geom)), p1.geom) -- << We cannot find a value to group on that more then two elements have in common
```

Two things I haven't explored yet are so called recursive queries. If the first code would be written recursively (joining the table itself till no new joins occur) it could be a solution.

The second idea I haven't explored would be using `ST_ClusterIntersecting`, though I'm not sure if it is of any help here.


As far as I can tell your query should be able to handle merging more then 2 elements. So as a third potential solution we could just use the first value that comes out of the group by class. There are a lot of [solutions](https://stackoverflow.com/questions/3800551/select-first-row-in-each-group-by-group). I've never used [partitioning](https://stackoverflow.com/a/3800572/3771196) maybe it could help as well.